pkg/operator/ceph/webhook-config.go

```diff
@@ -0,0 +1,269 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	v1 "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	admv1 "k8s.io/api/admissionregistration/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	issuerName        = "selfsigned-issuer"
+	certificateName   = "rook-admission-controller-cert"
+	webhookConfigName = "rook-ceph-webhook"
+)
+
+func createIssuer(ctx context.Context, webhookContext *WebhookContext) (*api.Issuer, error) {
+	logger.Infof("Fetching Issuer %s/%s.", namespace, issuerName)
+	issuer, err := webhookContext.Admission.Issuers(namespace).Get(ctx, issuerName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Infof("Creating Issuer %s/%s.", namespace, issuerName)
```

Comment:
How about factoring this code block out to a helper method to create the issuer?

Reply:
refactored